### PR TITLE
Add Open Collective social icon

### PR DESCRIPTION
**Pull request by Pukimaa**
#### Description
This PR aims to add an icon of [Open Collective](https://opencollective.com) to the Icon Component. Open Collective is a crowdfunding platform also used by Astro itself, so it would only make sense to be added, right?

Preview of the openCollective Icon

### CODE REVIEW

File: .changeset/mighty-ways-marry.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/starlight': patch
+---
+
+Add Open Collective social icon
```
Comment (at-the-vr) [inline .changeset/mighty-ways-marry.md:5]: Minor typo: update the changeset more to present tense? Reference: [Astro Docs Docs](https://contribute.docs.astro.build/guides/new-feature-docs/#changesets)
```suggestion
Adds Open Collective social link icon
```
Icon is working 💯 on [Preview](https://starlight-d5nl6x0n4-astrodotbuild.vercel.app/guides/components/#all-icons)

**Review by at-the-vr — APPROVED** [review]
LGTM ✅

**Review by ElianCodes — APPROVED** [review]
It indeed makes sense to me to add an Open Collective icon! Thanks for your help!

File: packages/starlight/components/Icons.ts [hunk 2]
```diff
@@ -103,4 +103,6 @@ export const Icons = {
...
+	openCollective:
+		'<path fill-rule="evenodd" clip-rule="evenodd" d="M21.8649 5.16556C23.2111 7.10496 24 9.46035 24 12C24 14.5397 23.2111 16.895 21.8649 18.8344L18.7564 15.726C19.3668 14.6214 19.7143 13.3513 19.7143 12C19.7143 10.6487 19.3668 9.37857 18.7564 8.27403L21.8649 5.16556ZM18.8344 2.1351L15.726 5.24357C14.6214 4.63315 13.3513 4.28571 12 4.28571C7.73952 4.28571 4.28571 7.73952 4.28571 12C4.28571 16.2605 7.73952 19.7143 12 19.7143C13.3513 19.7143 14.6214 19.3668 15.726 18.7564L18.8344 21.8649C16.895 23.2111 14.5397 24 12 24C5.37258 24 0 18.6274 0 12C0 5.37258 5.37258 0 12 0C14.5397 0 16.895 0.788937 18.8344 2.1351Z"/><path fill-rule="evenodd" clip-rule="evenodd" d="M21.8649 5.16556C23.2111 7.10496 24 9.46035 24 12C24 14.5397 23.2111 16.895 21.8649 18.8344L18.7564 15.726C19.3668 14.6214 19.7143 13.3513 19.7143 12C19.7143 10.6487 19.3668 9.37857 18.7564 8.27403L21.8649 5.16556Z"/>'
```
Comment (delucis) [inline packages/starlight/components/Icons.ts:107]: Ran the SVG through [SVGOMG](https://jakearchibald.github.io/svgomg/) to compress it a little and reduce the size:

```suggestion
		'<path d="M21.86 5.17a11.94 11.94 0 0 1 0 13.66l-3.1-3.1a7.68 7.68 0 0 0 0-7.46l3.1-3.1Zm-3.03-3.03-3.1 3.1a7.71 7.71 0 1 0 0 13.51l3.1 3.11a12 12 0 1 1 0-19.73Z"/><path d="M21.86 5.17a11.94 11.94 0 0 1 0 13.66l-3.1-3.1a7.68 7.68 0 0 0 0-7.46l3.1-3.1Z"/>'
```

**Review by delucis — COMMENTED** [review]
Thanks @Pukimaa — looks great! Left one small suggestion.

Could you also update branch? Looks like some checks are failing because it’s out of sync with `main`.

**Comment by Pukimaa** [comment]
Sure @delucis! We should be good to go now ^^

**Review by delucis — APPROVED** [review]
Thanks for your patience over the holidays @Pukimaa. This looks good to me — will aim to get a patch release out with this later today 🙌

**Pull request merged**